python/dgl/_ffi/streams.py

```diff
@@ -63,3 +58,43 @@ def stream(cuda_stream):
         target stream will be set.
     """
     return StreamContext(cuda_stream)
+
+def to_dgl_stream_handle(cuda_stream):
+    """ Convert torch.cuda.Stream to DGL stream handle
+
+    Parameters
+    ----------
+    stream : torch.cuda.Stream.
+
+    Returns
+    -------
+    DGLStreamHandle.
+    """
+    return ctypes.c_void_p(cuda_stream.cuda_stream)
+
+def _dgl_set_stream(ctx, cuda_stream):
+    """ Set the current CUDA stream of the given DGL context.
+    Use of this function is discouraged in favor of torch's stream context.
+
+    Parameters
+    ----------
+    ctx: DGLContext
+    stream: DGLStreamHandle
+    """
+    check_call(_LIB.DGLSetStream(ctx.device_type, ctx.device_id, cuda_stream))
```

Comment:
DGLSetStream needs to be fixed, as now it still fetches `CUDAThreadEntry::ThreadLocal()->stream`

Reply:
I'm in favor of removing `SetStream`, `GetStream` APIs and DGL's `stream()` context, at least at the Python level, because they have been replaced by PyTorch's APIs.

Reply:
I am Okay with removing `SetStream`, `GetStream` in Python level. 
In fact, now I think I am also fine with removing  `SetStream`/`CreateStream` in C++ level, if only single stream (current pytorch stream) is used for now in DGL. We can add it back when necessary in future.

Reply:
> My main concern is whether we would have imported too many helper functions from the TensorAdaptor if we redirected `SetStream`/`GetStream` to PyTorch.

This is OK.  TensorAdapter is just an interface to call PyTorch C++ functions in DGL, so you can have as many functions as you want.